hardware_interface/README.md

```diff
@@ -0,0 +1,299 @@
+    jnt_pos_interface.registerHandle(pos_handle_a);
+
+    hardware_interface::JointHandle pos_handle_b(jnt_state_interface.getHandle("B"), &cmd[1]);
+    jnt_pos_interface.registerHandle(pos_handle_b);
+
+    // Register the JointPositionInterface containing the read/write joints
+    // with this robot's hardware_interface::RobotHW.
+    registerInterface(&jnt_pos_interface);
+
+    return true;
+  }
+
+private:
+  hardware_interface::JointStateInterface jnt_state_interface;
+  hardware_interface::PositionJointInterface jnt_pos_interface;
+
+  // Data member array to store the controller commands which are sent to the 
+  // robot's resources (joints, actuators)
+  double cmd[2];
+
+  // Data member arrays to store the state of the robot's resources (joints, sensors)
+  double pos[2];
+  double vel[2];
+  double eff[2];
+};
+```
+
+This code represents a custom robot and is required to control it. 
+The functions above are designed to give the controller manager (and the 
+controllers inside the controller manager) access to the joint state of a
+custom robot, and to command it. When the controller manager runs, the 
+controllers will read from the pos, vel and eff variables of the custom robot 
+hardware interface, and the controller will write the desired command into the 
+cmd variable. It's mandatory to make sure the pos, vel and eff variables always 
+have the latest joint state available, and to make sure that whatever is written
+into the cmd variable gets executed by the robot. This can be done by implementing 
+\ref hardware_interface::RobotHW::read() and a \ref hardware_interface::RobotHW::write()
+methods. A node's `main()` function can be implemented like this:
+
+```cpp
+#include <ros/ros.h>
+#include <my_robot/my_robot.h>
+#include <controller_manager/controller_manager.h>
+
+int main(int argc, char **argv)
+{
+    // Initialize the ROS node
+    ros::init(argc, argv, "my_robot");
+    
+    // Create an instance of your robot so that this instance knows about all 
+    // the resources that are available.
+    MyRobot::MyRobot robot;
+
+    // Create an instance of the controller manager and pass it the robot, 
+    // so that it can handle its resources.
+    controller_manager::ControllerManager cm(&robot);
+    
+    // Setup a separate thread that will be used to service ROS callbacks.
+    ros:AsyncSpinner spinner(1);
+    spinner.start();
+    
+    // Setup for the control loop.
+    ros::Time prev_time = ros::Time::now();
+    ros::Rate rate(10.0); // 10 Hz rate
+    
+    while (ros::ok())
+    {
+        // Basic bookkeeping to get the system time in order to compute the control period.
+        const ros::Time     time   = ros:Time::now();
+        const ros::Duration period = time - prev_time;
+        
+        // Execution of the actual control loop.
+        robot.read();
+        cm.update(time, period);
+        root.write();
```

Comment:
Where exactly in the control loop should we use the `transmission_interface` if a robot's transmissions need to be defined in software? Is this done inside `read()` and `write()` methods? Or is this already handled in the URDF definition (at least for the available transmissions such as [`SimpleTransmission`](http://docs.ros.org/melodic/api/transmission_interface/html/c++/classtransmission__interface_1_1SimpleTransmission.html)?